packages/aws-cdk/lib/settings.ts

```diff
@@ -234,10 +234,21 @@ export class Settings {
     return context;
   }
 
-  private static parseStringTagsListToObject(argv: Arguments): Tag[] {
+  /**
+   * Parse tags out of arguments
+   *
+   * Return undefined if no tags were provided, return an empty array if only empty
+   * strings were provided
+   */
+  private static parseStringTagsListToObject(argTags: string[] | undefined): Tag[] | undefined {
+    if (argTags === undefined) { return undefined; }
+    if (argTags.length === 0) { return undefined; }
+    const nonEmptyTags = argTags.filter(t => t !== '');
+    if (nonEmptyTags.length === 0) { return []; }
```

Comment:
Is the input to this function just the input from the cli (`cli.ts`), or could it also be from an existing stack? I'm trying to wrap my head around if passing the CLI tags through `arrayFromYargs` would effectively de-dupe this logic.

Reply:
I know! I wanted to do that originally but the types of doing that don't really work out.

It doesn't sit well with me where this parsing is done, but I didn't want to change the existing code too much more because the diff was already growing too large for my tastes.